geopandas/tests/test_sindex.py

```diff
@@ -172,10 +174,309 @@ def test_intersection_invalid_bounds_tuple(self, test_geom):
                 # rtree raises an RTreeError which we need to catch
                 self.df.sindex.intersection(test_geom)
 
+    # ------------------------------ `query` tests ------------------------------ #
+    @pytest.mark.parametrize(
+        "predicate, test_geom, expected",
+        (
+            (None, (-1, -1, -0.5, -0.5), []),
+            (None, (-0.5, -0.5, 0.5, 0.5), [0]),
+            (None, (0, 0, 1, 1), [0, 1]),
+            ("intersects", (-1, -1, -0.5, -0.5), []),
+            ("intersects", (-0.5, -0.5, 0.5, 0.5), [0]),
+            ("intersects", (0, 0, 1, 1), [0, 1]),
+            ("within", (0.25, 0.28, 0.75, 0.75), []),  # does not intersect
+            ("within", (0, 0, 10, 10), []),  # intersects but is not within
+            ("within", (11, 11, 12, 12), [5]),  # intersects and is within
+            ("contains", (0, 0, 1, 1), []),  # intersects but does not contain
+            ("contains", (0, 0, 1.001, 1.001), [1]),  # intersects and contains
+            ("contains", (0.5, 0.5, 1.5, 1.5), [1]),  # intersects and contains
+            ("contains", (-1, -1, 2, 2), [0, 1]),  # intersects and contains multiple
+        ),
+    )
+    def test_query(self, predicate, test_geom, expected):
+        """Tests the `query` method with valid inputs and valid predicates."""
+        test_geom = box(*test_geom)
+        res = self.df.sindex.query(test_geom, predicate=predicate)
+        assert_array_equal(res, expected)
+
+    @pytest.mark.parametrize(
+        "test_geom, expected_error", [("notavalidgeom", TypeError)],
+    )
+    def test_query_invalid_geometry(self, test_geom, expected_error):
+        """Tests the `query` method with invalid geometry.
+        """
+        with pytest.raises(expected_error):
+            self.df.sindex.query(test_geom)
+
+    @pytest.mark.parametrize(
+        "test_geom, expected_error, expected_value",
+        [
+            (None, None, []),
+            (GeometryCollection(), None, []),
+            (Point(), None, []),
+            (MultiPolygon(), None, []),
+            (Polygon(), None, []),
+        ],
+    )
+    def test_query_empty_geometry(self, test_geom, expected_error, expected_value):
+        """Tests the `query` method with empty geometry.
+        """
+        if expected_error is not None:
```

Comment:
Is there going to be a case where you actually expect an error (now this is always None), otherwise this can be cleaned up a bit?

Reply:
I think that can be removed in some tests, I copy pasted and wasn't sure if it would be needed or not. I'll go through and remove where it is not being used, I guess it can always be added back later.

Reply:
Parameter removed in next commit.